habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -363,6 +367,15 @@ def parse_receptacles_from_user_config(
             )
             receptacle_scale = ao_uniform_scaling * sub_config.get("scale")
 
+            # TODO: parse this from user config
+            category = re.sub(
```

Comment:
We are anyways saving the `parent_object_handle` in the regular rearrange dataset. Can't we just use this information and not have to change the dataset format for this object rearrange task? I would rather make no changes to the dataset code and have extracting the category be on the task side. Then we don't have to worry about managing different dataset instances.

Reply:
This particular change actually just introduces an optional "category" field for the receptacles in the scene. It does not affect the dataset format. But I did make changes to the dataset format.

**On dataset format**
The [newly added fields](https://github.com/yvsriram/habitat-lab/blob/6429604ba17b2f1d17d004c8e036c27b59bf101f/habitat-lab/habitat/datasets/rearrange/rearrange_dataset.py#L55) are goal specifications (object and receptacle category names) and possible solutions (valid object and receptacles - in the format ObjectNav uses). Currently, I am adding these in a [separate script](https://github.com/yvsriram/lang-rearrange-scripts/blob/master/datasets/add_cat_fields_to_episodes.py) because I thought we would handle this as part of the dataset generation. But this code can definitely be pulled into the task specification. 

**Why a new dataset format may be more appropriate?**
- This would allow the episodes to be modified flexibly. 
    - We can support a range of tasks by only modifying the object goal and possible candidate objects (solutions). For example, in the future, we can directly handle goal specification that includes object attributes like "red colored bottle".
    - We may want to preprocess and store the valid goal viewpoints (for navigation skills) in episodes similar to ObjectNav.
- The new fields can be optional (as they are currently) allowing the new ObjectRearrangeDataset to be also used for loading geometric rearrange datasets. So we needn't worry about handling different dataset instances.
- I feel category extraction for objects/receptacles is better handled independently of the task. We actually may not be extracting the receptacle category from handle names and may use custom categories.


So I think for now, we can continue to generate GeoGoal rearrange episodes and use a separate script to append fields needed for the task. However, we should quickly add these changes to the dataset generation script as well. Let me know if you think otherwise.

Reply:
Thanks for the explanation. I agree with what you said. It makes sense to have this as a separate script as a "geometric rearrange dataset add on". Nothing to change here then. Your current code is good for this. Just be sure you document these steps of creating the dataset somewhere since this is a new way to create the dataset.